### PR TITLE
replaced | to , in metamask signature, calculate logo to sha256 

### PR DESCRIPTION
**Pull request by axelrod-blox**
Screenshot 2023-07-26 at 15 13 07

### CODE REVIEW

File: src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx [hunk 1]
```diff
@@ -47,10 +48,16 @@ const EditOperatorDetails = () => {
         if (!isNotValidity) {
             let payload = metadataStore.createMetadataPayload();
             let rawDataToValidate: any = [];
-            Object.values(payload).map(value =>
-                rawDataToValidate.push(value),
+            Object.keys(payload).map(key => {
+                if (key === FIELD_KEYS.OPERATOR_IMAGE && payload[key]){
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx]: done

File: src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx [hunk 2]
```diff
@@ -47,10 +48,16 @@ const EditOperatorDetails = () => {
         if (!isNotValidity) {
             let payload = metadataStore.createMetadataPayload();
             let rawDataToValidate: any = [];
-            Object.values(payload).map(value =>
-                rawDataToValidate.push(value),
+            Object.keys(payload).map(key => {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx]: done

**Comment by meshin-dev** [comment]
@axelrod-blox I don't see `logo:sha256:...` prefix in data that metamask receives, on your screenshot

File: src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx [hunk 3]
```diff
@@ -47,10 +48,14 @@ const EditOperatorDetails = () => {
...
-            );
-            rawDataToValidate = rawDataToValidate.join('|');
+            Object.keys(payload).forEach(field => {
+                if (payload[field]) {
+                    const newItem =
+                        field === 'logo' ? sha256(payload[field]) : payload[field];
+                    rawDataToValidate.push(newItem);
+                }
+            });
+            rawDataToValidate = rawDataToValidate.join(',');
```
Comment (meshin-dev) [inline src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx:58]: This is not correct implementation.
In requirements was not anything about removing sequential compilation of the data with filed name array.

**Review by meshin-dev — CHANGES_REQUESTED** [review]

File: src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx (hunk 3 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx:58]: done

Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/EditOperatorDetails/EditOperatorDetails.tsx:58]: done

**Review by meshin-dev — APPROVED** [review]

**Pull request merged**